pkg/autoscaler/autoscaler/util.go

```diff
@@ -57,11 +59,81 @@ func checkStsAutoScalingPrerequisites(set *appsv1.StatefulSet) bool {
 	return true
 }
 
-// checkStsAutoScalingInterval would check whether there is enough interval duration between every two auto-scaling
-func checkStsAutoScalingInterval(tac *v1alpha1.TidbClusterAutoScaler, intervalSeconds int32, memberType v1alpha1.MemberType) (bool, error) {
+func checkStsAutoScaling(tac *v1alpha1.TidbClusterAutoScaler, thresholdSeconds, intervalSeconds int32, memberType v1alpha1.MemberType) (bool, error) {
+	realClock := clockwork.NewRealClock()
+	if tac.Annotations == nil {
+		tac.Annotations = map[string]string{}
+	}
+	// 3*controller.ResyncDuration is maximum time allowed before reset phase status
+	ableToScale, err := checkStsLastAutoScalingTimestamp(tac, 3*controller.ResyncDuration, realClock)
+	if err != nil {
+		return false, err
+	}
+	if !ableToScale {
+		return false, nil
+	}
+	ableToScale, err = checkStsReadyAutoScalingTimestamp(tac, thresholdSeconds, realClock)
+	if err != nil {
+		return false, err
+	}
+	if !ableToScale {
+		return false, nil
+	}
+	ableToScale, err = checkStsAutoScalingInterval(tac, intervalSeconds, memberType)
+	if err != nil {
+		return false, err
+	}
+	if !ableToScale {
+		return false, nil
+	}
+	return true, nil
+}
+
+// checkStsLastAutoScalingTimestamp reset TiKV phase if last auto scaling timestamp is longer than thresholdSec
+func checkStsLastAutoScalingTimestamp(tac *v1alpha1.TidbClusterAutoScaler, thresholdSec time.Duration, clock clockwork.Clock) (bool, error) {
 	if tac.Annotations == nil {
 		tac.Annotations = map[string]string{}
 	}
+
+	lastAutoScalingTimestamp, existed := tac.Annotations[label.AnnTiKVLastAutoScalingTimestamp]
```

Comment:
`AnnTiKVLastAutoScalingTimestamp` is used to check the interval duration between each 2 auto-scaling, not syncing. so I tend to use annotation like `last-syncing` to record the timestamp for each syncing, and check the interval duration between each 2 syncing whether is meet the requirement.